lightningd/peer_control.c

```diff
@@ -112,6 +113,9 @@ struct peer *new_peer(struct lightningd *ld, u64 dbid,
 	peer->connected = PEER_DISCONNECTED;
 	peer->last_connect_attempt.ts.tv_sec
 		= peer->last_connect_attempt.ts.tv_nsec = 0;
+	if (their_features)
+		peer_update_features(peer, their_features);
+
```

Comment:
Rather than initializing to NULL (line 110) then overriding, do the obvious:

```
if (their_features)
    peer->their_features = tal_dup_talarr(peer, u8, their_features);
else
    peer->their_features = NULL;
```
(There's no tal_dup_talarr_or_null fn...).

Reply:
done, removed `peer_update_features` as its now unused